src/main/java/io/github/christophermanahan/carnitas/Serializer.java

```diff
@@ -0,0 +1,29 @@
+package io.github.christophermanahan.carnitas;
+
+import java.util.List;
+
+class Serializer {
+    static final String CRLF = "\r\n";
+    static final String BLANK_LINE = "\r\n\r\n";
+
+    byte[] serialize(HTTPResponse response) {
+        return (
+          statusLine(response)
+          + format(response.headers())
+          + BLANK_LINE
+          + body(response)
+        ).getBytes();
+    }
+
+    private String statusLine(HTTPResponse response) {
+        return HTTPResponse.VERSION + " " + response.status().code + Serializer.CRLF;
+    }
+
+    private String format(List<String> headers) {
+        return String.join(CRLF, headers);
+    }
+
+    private String body(HTTPResponse response) {
+        return response.body().orElse("");
+    }
+}
```

Comment:
This is great! Serialization is definitely a responsibility that should have its own class.

Reply:
Yeah, this definitely cleaned up a lot of the tests.